lib/app.js

```diff
@@ -9,12 +9,13 @@ const ErrorDebug = process.env.NODE_ENV === 'production'
 
 export default class App extends Component {
   static childContextTypes = {
-    headManager: PropTypes.object
+    headManager: PropTypes.object,
+    router: PropTypes.object
   }
 
   getChildContext () {
-    const { headManager } = this.props
-    return { headManager }
+    const { headManager, router } = this.props
```

Comment:
I'm not sure which fields we should expose and other details of how this should be implemented. Do you want to take over from here?